k9mail-library/src/test/java/com/fsck/k9/mail/internet/MimeMessageParseTest.java

```diff
@@ -272,4 +273,60 @@ private static void checkLeafParts(MimeMessage msg, String... expectedParts) thr
         }
         assertEquals(Arrays.asList(expectedParts), actual);
     }
+
+    @Test
+    public void isXOriginalTo() throws Exception {
+        MimeMessage msg = parseWithoutRecurse(toStream(
+                "From: <adam@example.org>\r\n" +
+                        "To: <eva@example.org>\r\n" +
+                        "X-Original-To: <test@mail.com>\r\n"+
+                        "Subject: Testmail\r\n" +
+                        "MIME-Version: 1.0\r\n" +
+                        "Content-type: text/plain\r\n" +
+                        "Content-Transfer-Encoding: 7bit\r\n" +
+                        "\r\n" +
+                        "this is some test text."));
+
+        Address address = new Address("<test@mail.com>");
+        Address[] x_original_address = msg.getRecipients(RecipientType.X_ORIGINAL_TO) ;
+        assertTrue(x_original_address[0].equals(address));
```

Comment:
If this test fails the error message generated by JUnit won't be too helpful. Change to the following instead:

```java
assertEquals(new Address("<test@mail.com>"), xOriginalAddresses[0]);
```

Before that you should probably check the number of returned addresses.

```java
assertEquals(1, xOriginalAddresses.length);
```